src/Credentials/GCECredentials.php

```diff
@@ -120,25 +128,37 @@ public static function onGce(ClientInterface $client = null)
    * Implements FetchAuthTokenInterface#fetchAuthToken.
    *
    * Fetches the auth tokens from the GCE metadata host if it is available.
-   * If $client is not specified a new GuzzleHttp\Client instance is used.
+   * If $httpHandler is not specified a the default HttpHandler is used.
    *
-   * @param $client GuzzleHttp\ClientInterface optional client.
+   * @param callable $httpHandler callback which delivers psr7 request
    * @return array the response
    */
-  public function fetchAuthToken(ClientInterface $client = null)
+  public function fetchAuthToken(callable $httpHandler = null)
   {
-    if (is_null($client)) {
-      $client = new Client();
+    if (is_null($httpHandler)) {
+      $httpHandler = HttpHandlerFactory::build();
     }
     if (!$this->hasCheckedOnGce) {
-      $this->isOnGce = self::onGce($client);
+      $this->isOnGce = self::onGce($httpHandler);
     }
     if (!$this->isOnGce) {
       return array();  // return an empty array with no access token
     }
-    $resp = $client->get(self::getTokenUri(),
-                         [ 'headers' => [self::FLAVOR_HEADER => 'Google']]);
-    return $resp->json();
+    $resp = $httpHandler(
+      new Request(
+        'GET',
+        self::getTokenUri(),
+        [self::FLAVOR_HEADER => 'Google']
+      )
+    );
+    $body = (string) $resp->getBody();
+
+    // Assume it's JSON; if it's not throw an exception
+    if (null === $json = json_decode($body, true)) {
+      throw new \Exception('Invalid JSON response');
```

Comment:
yes, using [`json_last_error`](http://php.net/manual/en/function.json-last-error.php), but it's not super straightforward